code/modules/antagonists/changeling/abomination.dm

```diff
@@ -0,0 +1,95 @@
+/*
+	- CHANGELING ABOMINATION -
+
+	Contents:
+		Reagent
+		Monster
+*/
+
+/datum/reagent/shiftium
+	name = "shifium"
+	description = "A changeling matter derived stimulant that overexcites the nervous system, forcing transformations in changelings."
+	color = "#9D5A99"
+	overdose_threshold = 8
+	taste_description = "savory with a bit of blood"
+	metabolization_rate = 0.3 * REAGENTS_METABOLISM
+	var/obj/shapeshift_holder/shapeshiftdata
+
+/datum/reagent/shiftium/on_mob_metabolize(mob/living/L)
+	L.adjustStaminaLoss(-10, 0)
+	..()
+
+/datum/reagent/shiftium/on_mob_end_metabolize(mob/living/L)
+	..()
+	if(!shapeshiftdata)
+		return
+	shapeshiftdata.restore()
+	REMOVE_TRAIT(L, CHANGELING_HIVEMIND_MUTE, type)
+
+/datum/reagent/shiftium/overdose_start(mob/living/L)
+	if (L.mind?.has_antag_datum(/datum/antagonist/changeling))
+		to_chat(L,"<span class='danger'>You struggle to maintain your form!</span>")
+	..()
+
+/datum/reagent/shiftium/overdose_process(mob/living/L)
+	L.adjustStaminaLoss(5, 0)
+	if(prob(volume * 2.5))
```

Comment:
You need to add a var to stop this from passing multiple times.

Reply:
```suggestion
	if(!shapeshiftdata && prob(volume * 2.5))
```